addons/knowledge/static/src/js/widgets/knowledge_dialogs.js

```diff
@@ -0,0 +1,186 @@
+odoo.define('knowledge.dialogs', function (require) {
+'use strict';
+
+const Dialog = require('web.Dialog');
+const { _t } = require('web.core');
+
+const MoveArticleToDialog = Dialog.extend({
+    template: 'knowledge.knowledge_move_article_to_modal',
+    /**
+     * @override
+     * @param {Widget} parent
+     * @param {Object} options
+     * @param {Object} props
+     */
+    init: function (parent, options, props) {
+        // Set default options:
+        options.title = options.title || _t('Move an Article');
+        options.buttons = options.buttons || [{
+            text: _t('Ok'),
+            classes: 'btn-primary',
+            click: this.save.bind(this)
+        }, {
+            text: _t('Cancel'),
+            close: true
+        }];
+        this._super(...arguments);
+        this.props = props;
+        // Set template variables:
+        const { state } = props;
+        this.article_name = state.data.name;
+    },
+
+    /**
+     * @override
+     * @returns
+     */
+    start: async function () {
```

Comment:
It's usually not a good practice to place "await" inside a start override.
This means that is someone overrides your function, he cannot perform additional behaviors "at the same time" as the original start behavior.

Instead it's more flexible for overrides to do:

```js
return this._super(...arguments).then(() => {
    this.initSelect2();
});
```